File: src/main/java/fr/lewon/web/bot/entities/input/Action.java
package fr.lewon.web.bot.entities.input;

import fr.lewon.bot.http.body.urlencoded.FUEMember;

public abstract class Action {

	@FUEMember("class")
	private String clazz;

	@FUEMember
	private String action;

	public Action(String clazz, String action) {
		this.clazz = clazz;
		this.action = action;
	}

	public String getClazz() {
		return clazz;
	}

	public void setClazz(String clazz) {
		this.clazz = clazz;
	}

	public String getAction() {
		return action;
	}

	public void setAction(String action) {
		this.action = action;
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/Action.kt
package fr.lewon.bot.hh.entities.input

import com.fasterxml.jackson.annotation.JsonProperty

abstract class Action(clazz: String, action: String?) {

    @JsonProperty("class")
    val clazz: String = clazz

    @JsonProperty
    var action: String? = action

}